middleware/checkAuth.js

```diff
@@ -0,0 +1,85 @@
+const jwt = require('jsonwebtoken');
+const ERROR = require('../constants').ERROR;
+const User = require('../models/User');
+
+const TOKEN_EXPIRED_ERROR = 'TokenExpiredError';
+
+module.exports = async (req, res, next) => {
+  if (req.method === 'OPTIONS') {
+    return next();
+  }
+
+  let user;
+  let accessToken;
+  let decodedAccessToken;
+  let refreshToken;
+  let decodedRefreshToken;
+
+  try {
+    accessToken = req.headers.authorization.split(' ')[1];
+
+    decodedAccessToken = jwt.verify(accessToken, process.env.JWT_SECRET_KEY);
+
+    const { id, nickname, imageUrl } = decodedAccessToken;
+
+    user = await User.findById(id);
+
+    refreshToken = user.refreshToken;
+
+    try {
+      decodedRefreshToken = jwt.verify(user.refreshToken, process.env.JWT_SECRET_KEY);
```

Comment:
`refreshToken` 변수를 활용하거나, 이 변수를 없애야 될 것 같습니다.
```suggestion
    refreshToken = user.refreshToken;

    try {
      decodedRefreshToken = jwt.verify(refreshToken, process.env.JWT_SECRET_KEY);
```